tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -0,0 +1,876 @@
+}
+
+// returnNicIpdetails returns ipName by taking instance name as an argument.
+func (az *AzureProvider) getIPname(ctx context.Context, vmName string) (err error, ipName string) {
+	vmStruct, err := az.vmClient.Get(ctx, az.resourceGroupName, vmName, "instanceView")
+	if err != nil {
+		log.Error(err, fmt.Sprintf("cannot fetch the instance data of %s", vmName))
+		return
+	}
+	networkProfile := vmStruct.VirtualMachineProperties.NetworkProfile
+	networkInterface := (*networkProfile.NetworkInterfaces)[0]
+	nicID := *networkInterface.ID
+	nicName := extractResourceName(nicID)
+	interfaceStruct, err := az.nicClient.Get(ctx, az.resourceGroupName, nicName, "")
+	if err != nil {
+		log.Error(err, fmt.Sprintf("cannot fetch the network interface data of %s", vmName))
+	}
+	interfacePropFormat := *(interfaceStruct.InterfacePropertiesFormat)
+	interfaceIPConfigs := *(interfacePropFormat.IPConfigurations)
+	ipConfigProp := *(interfaceIPConfigs[0].InterfaceIPConfigurationPropertiesFormat)
+	ipID := *(ipConfigProp.PublicIPAddress.ID)
+	ipName = extractResourceName(ipID)
+	return nil, ipName
+}
+
+// destroyIP deletes the IP address by taking it's name as argument.
+func (az *AzureProvider) destroyIP(ctx context.Context, ipName string) (err error) {
+	_, err = az.ipClient.Delete(ctx, az.resourceGroupName, ipName)
+	if errorCheck(err) {
+		log.Error(err, fmt.Sprintf("failed to delete the public IP: %s", ipName))
+		return
+	}
+	return
+}
+
+// destroyInstance deletes the instance by taking it's name as argument.
+func (az *AzureProvider) destroyInstance(ctx context.Context, vmName string) (err error) {
+	future, err := az.vmClient.Delete(ctx, az.resourceGroupName, vmName)
+	if errorCheck(err) {
+		log.Error(err, fmt.Sprintf("failed to delete the instance: %s", vmName))
+		return
+	}
+	err = future.WaitForCompletionRef(ctx, az.vmClient.Client)
+	if errorCheck(err) {
+		log.Error(err, fmt.Sprintf("failed to delete the instance: %s", vmName))
+		return
+	}
+	return nil
+}
+
+// destroyNIC deletes the nic by taking it's name as argument.
+func (az *AzureProvider) destroyNIC(ctx context.Context, nicName string) (err error) {
+	_, err = az.nicClient.Delete(ctx, az.resourceGroupName, nicName)
+	if errorCheck(err) {
+		log.Error(err, fmt.Sprintf("failed to delete the nic: %s", nicName))
+		return
+	}
+	return
+}
+
+// deleteSpecificRule deletes particular rule set by name from the list of available rules set.
+func (az *AzureProvider) deleteSpecificRule(s []network.SecurityRule, name string) (updatedList []network.SecurityRule) {
+	for i, rule := range s {
+		if !checkForNil(rule.Name) {
+			if strings.Contains(*(rule.Name), name) {
+				updatedList = append(s[:i], s[i+1:]...)
+			}
+		}
+	}
+	return
+}
+
+// destroyNSG deletes the security group rules by taking it's name as argument.
+// it deletes the rdp and vnet_traffic rules from the worker subnet security group rules.
+func (az *AzureProvider) updateNSGRules(ctx context.Context, nsgName string) (err error) {
```

Comment:
Doc mismatch

Reply:
The function should match what it is doing. The mismatch here was it should be called `deleteNSGRules`. You are not updating any rules here but deleting the rules from a SG.